src/test/java/bio/terra/app/controller/DatasetsApiControllerTest.java

```diff
@@ -240,6 +240,38 @@ void testUpdateSnapshotBuilderSettings() throws Exception {
   }
 
   @Test
+  void testRequestSnapshot() throws Exception {
+    mockValidators();
+    JobModel expected =
+        new JobModel()
+            .id("id")
+            .description("Stub Method")
+            .jobStatus(JobModel.JobStatusEnum.SUCCEEDED)
+            .statusCode(200)
+            .completed("completed")
+            .submitted("submitted")
+            .className("SnapshotAccessRequest");
+    SnapshotAccessRequest input =
+        new SnapshotAccessRequest()
+            .name("name")
+            .researchPurposeStatement("purpose")
+            .datasetRequest(new SnapshotRequest());
+    when(snapshotBuilderService.requestSnapshot(DATASET_ID, input)).thenReturn(expected);
+    String actualJson =
+        mvc.perform(
+                post(REQUEST_SNAPSHOT_ENDPOINT, DATASET_ID)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(TestUtils.mapToJson(input)))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    JobModel actual = TestUtils.mapFromJson(actualJson, JobModel.class);
+    assertThat("The request succeeded", actual, equalTo(expected));
```

Comment:
Nit: For the message here, I would say "The request returned the {RESPONSE OBJECT EXPECTED}".

I think the request succeeding test is actually on line 265 with `.andExpect(status().isOk())`